module_utils/network_lsr/argument_validator.py

```diff
@@ -896,6 +896,14 @@ def _validate_post_fields(self, value, name, result):
         state = result.get("state")
         persistent_state = result.get("persistent_state")
 
+        # if persistent state is present state cannot be set to present or absent
+        if persistent_state == "present" and state:
+            if state in ("absent", "present"):
+                raise ValidationError(
+                    name,
+                    "State cannot be '{}' if persistent_state is present".format(state),
```

Comment:
How about making this postive like "state can only be up or down if persistent_state is ....". Also, using `.format()` with  `{}` is not Python 2.6 compatible if I recall correctly.

Reply:
Not sure, why this is not failing in the unit test...

Reply:
Fixed, thanks!